decoderx/http.go

```diff
@@ -335,20 +335,11 @@ func (t *HTTP) decodeJSONForm(r *http.Request, destination interface{}, o *httpD
 	}
 
 	values := url.Values{}
-	var notJSONForm bool
 	parsed.ForEach(func(k, v gjson.Result) bool {
-		if v.IsArray() || v.IsObject() {
-			notJSONForm = true
-			return false
-		}
 		values.Set(k.String(), v.String())
 		return true
 	})
 
-	if notJSONForm {
-		return t.decodeJSON(r, destination, o, true)
-	}
```

Comment:
This piece of code prevents a "mixed" scenario where part of the input JSON isn't flattened like so:
```json
{
  "dynamic_object": {
    "stuff": { "blub": [42, 3.14152], "fu": "bar" },
    "consent": true
  },
  "name.first": "Horst",
  "name.last": "Horstmann"
}
```
I removed the defaulting to `decodeJSON()` completely.
I added some tests around this and also no existing tests were failing, so I think we are good.

Reply:
We have option `expectJSONFlattened` which is set by:

```
// HTTPDecoderJSONFollowsFormFormat if set tells the decoder that JSON follows the same conventions
// as the form decoder, meaning `{"foo.bar": "..."}` is translated to `{"foo": {"bar": "..."}}`.
func HTTPDecoderJSONFollowsFormFormat() HTTPDecoderOption {
	return func(o *httpDecoderOptions) {
		o.expectJSONFlattened = true
		o.keepRequestBody = true
	}
}
```

if that is set, we trigger `decodeJSONForm` but that in turn triggers regular JSON decoding if we encouter a complex type in the payload. This was introduced as part of ed839286 which mentions explicitly `fix(decoderx): improve form type detection`. I wasn't able to quickly find what caused this commit in Ory Kratos (probably some type of bug) but it would be good to understand the original reasoning behind this particular behavior (maybe it was a bug report for a specific edge case).